source/common/http/http2/codec_impl.cc

```diff
@@ -286,6 +316,26 @@ void ConnectionImpl::StreamImpl::resetStreamWorker(StreamResetReason reason) {
   ASSERT(rc == 0);
 }
 
+MetadataEncoder& ConnectionImpl::StreamImpl::getMetadataEncoder() {
+  if (metadata_encoder_ == nullptr) {
+    metadata_encoder_ = std::make_unique<MetadataEncoder>();
+  }
+  return *metadata_encoder_;
+}
+
+MetadataDecoder& ConnectionImpl::StreamImpl::getMetadataDecoder() {
+  if (metadata_decoder_ == nullptr) {
+    auto cb =
+        std::bind(&ConnectionImpl::StreamImpl::onMetadataDecoded, this, std::placeholders::_1);
```

Comment:
nit: I find bind hard to read, can you just pass an inline lambda / std::function?